File: baekjoon/rust/north_west_wind_5419.rs
use std::collections::{BTreeMap, HashMap, HashSet};
use std::io::{stdin, stdout, BufRead, BufReader, BufWriter, Write};

struct BIT {
    n: isize,
    bit: Vec<isize>,
}

impl BIT {
    fn new(n: isize) -> BIT {
        BIT {
            n: n,
            bit: vec![0; (n + 1) as usize],
        }
    }

    fn update(&mut self, mut p: isize, v: isize) {
        while p <= self.n {
            self.bit[p as usize] += v;
            p += p & (-p);
        }
    }

    fn _query(&self, mut b: isize) -> isize {
        let mut s = 0;
        while b > 0 {
            s += self.bit[b as usize];
            b -= b & (-b);
        }
        s
    }

    fn query(&self, a: isize, b: isize) -> isize {
        self._query(b) - self._query(a - 1)
    }
}

fn solution(n: i64, mut islands: BTreeMap<i64, Vec<i64>>, xs: HashMap<i64, i64>) -> String {
    let mut ans = 0;
    let mut bit = BIT::new((n + 1) as isize);

    for (_key, value) in islands.iter_mut() {
        value.sort_unstable();
        for x in value {
            ans += bit.query(1, *xs.get(&x).unwrap() as isize);
            bit.update(*xs.get(&x).unwrap() as isize, 1);
        }
    }
    String::from(ans.to_string())
}

fn main() {
    let mut reader = BufReader::new(stdin());
    let mut writer = BufWriter::new(stdout());
    let mut buffer = String::new();

    reader.read_line(&mut buffer).unwrap();
    let t = buffer.trim().parse::<i64>().unwrap();

    for _ in 0..t {
        buffer.clear();
        reader.read_line(&mut buffer).unwrap();
        let n = buffer.trim().parse::<i64>().unwrap();

        let mut xs_set = HashSet::<i64>::new();
        let mut xs = HashMap::<i64, i64>::new();
        let mut islands = BTreeMap::<i64, Vec<i64>>::new();

        for _ in 0..n {
            buffer.clear();
            reader.read_line(&mut buffer).unwrap();
            let pos = buffer
                .trim()
                .split_whitespace()
                .map(|x| x.parse::<i64>().unwrap())
                .collect::<Vec<_>>();
            xs_set.insert(pos[0]);
            if let Some(v) = islands.get_mut(&(-pos[1])) {
                v.push(pos[0])
            } else {
                islands.insert(-pos[1], vec![pos[0]]);
            }
        }
        let mut index = 1;
        let mut xs_vec = xs_set.into_iter().collect::<Vec<_>>();
        xs_vec.sort_unstable();
        for x in xs_vec {
            xs.insert(x, index);
            index += 1;
        }

        writeln!(writer, "{}", solution(n, islands, xs)).unwrap();
        writer.flush().unwrap();
    }
}

File: baekjoon/python/north_west_wind_5419.py
# Title: 북서풍
# Link: https://www.acmicpc.net/problem/5419

import sys
from collections import defaultdict

sys.setrecursionlimit(10 ** 6)


read_single_int = lambda: int(sys.stdin.readline().strip())
read_list_int = lambda: list(map(int, sys.stdin.readline().strip().split(' ')))

class BIT:
    def __init__(self, n):
        self.n = n
        self.bit = [0 for _ in range(n+1)]
    
    def update(self, p: int, v: int):
        while p <= self.n:
            self.bit[p] += v
            p += (p & (-p))

    def _query(self, b: int):
        s = 0
        while b > 0:
            s += self.bit[b]
            b -= (b & (-b))
        return s

    def query(self, a: int, b: int):
        return self._query(b) - self._query(a-1)


def solution(n: int, islands: defaultdict, xs: defaultdict):
    ans = 0
    bit = BIT(n+1)

    keys = sorted(islands.keys(), reverse=True)

    for key in keys:
        for x in sorted(islands[key]):
            ans += bit.query(1, xs[x])
            bit.update(xs[x], 1)

    return ans


def main():
    t = read_single_int()
    for _ in range(t):
        n = read_single_int()
        xs = defaultdict(lambda: -1)
        islands = defaultdict(lambda: [])
        for _ in range(n):
            x, y = read_list_int()
            xs[x] = 1
            islands[y].append(x)
        keys = sorted(xs.keys())
        for i, v in enumerate(keys, 1):
            xs[v] = i
        print(solution(n, islands, xs))


if __name__ == '__main__':
    main()